util/history_trimming_iterator.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#pragma once
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include "db/dbformat.h"
+#include "rocksdb/iterator.h"
+#include "rocksdb/slice.h"
+#include "table/internal_iterator.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class HistoryTrimmingIterator : public InternalIterator {
+ public:
+  HistoryTrimmingIterator(InternalIterator* input, const Comparator* cmp,
+                          const Slice& ts)
+      : input_(input), filter_ts_(ts), cmp_(cmp) {}
+
+  bool filter() {
+    if (!input_->Valid()) {
+      return true;
+    }
+    if (cmp_->timestamp_size() == 0) {
+      return true;
+    }
```

Comment:
Should we put this check in constructor? If ts is not enabled, just set valid_ to false, and status to `InvalidArgument()`?